base/reduce.jl

```diff
@@ -360,15 +360,19 @@ sum(a::AbstractArray{Bool}) = countnz(a)
 Returns the sum of all array elements, using the Kahan-Babuska-Neumaier compensated
 summation algorithm for additional accuracy.
 """
-function sum_kbn{T<:AbstractFloat}(A::AbstractArray{T})
+function sum_kbn(A::Any)
```

Comment:
There is no need for `::Any` here as no type assertions in the arguments of a function definition is effectively the same as using `::Any`